rm-datasource/src/main/java/io/seata/rm/datasource/exec/ExecuteTemplate.java

```diff
@@ -78,23 +80,7 @@
         if (sqlRecognizer == null) {
             executor = new PlainExecutor<>(statementProxy, statementCallback);
         } else {
-            switch (sqlRecognizer.getSQLType()) {
-                case INSERT:
-                    executor = new InsertExecutor<>(statementProxy, statementCallback, sqlRecognizer);
-                    break;
-                case UPDATE:
-                    executor = new UpdateExecutor<>(statementProxy, statementCallback, sqlRecognizer);
-                    break;
-                case DELETE:
-                    executor = new DeleteExecutor<>(statementProxy, statementCallback, sqlRecognizer);
-                    break;
-                case SELECT_FOR_UPDATE:
-                    executor = new SelectForUpdateExecutor<>(statementProxy, statementCallback, sqlRecognizer);
-                    break;
-                default:
-                    executor = new PlainExecutor<>(statementProxy, statementCallback);
-                    break;
-            }
+            executor = createSqlExecutor(sqlRecognizer.getSQLType(), sqlRecognizer, statementProxy, statementCallback);
```

Comment:
I think
```
createSqlExecutor(sqlRecognizer, statementProxy, statementCallback)
```
that is enough.

Reply:
@l81893521 Thank you for your advice

Reply:
@l81893521 Createsqlexecutor (sqlrecognizer, statementproxy, statementcallback) needs to return parameters to execute SQL for the executor

Reply:
I m apologize that I did not clear my point. I mean 3 parameters that s enough. just like that below.
```
executor = createSqlExecutor(sqlRecognizer, statementProxy, statementCallback);
```

Reply:
@l81893521 OK, I'll change it